pkgs/development/compilers/crystal/default.nix

```diff
@@ -1,11 +1,11 @@
 { stdenv, lib, fetchFromGitHub, fetchurl, makeWrapper
-, gmp, openssl, readline, tzdata, libxml2, libyaml
+, coreutils, git, gmp, nettools, openssl, readline, tzdata, libxml2, libyaml
 , boehmgc, libatomic_ops, pcre, libevent, libiconv, llvm, clang, which, zlib }:
 
 # We need multiple binaries as a given binary isn't always able to build
-# (even slightly) older or newer version.
+# (even slightly) older or newer versions.
 # - 0.26.1 can build 0.25.x and 0.26.x but not 0.27.x
-# - 0.27.2 can build 0.27.x but not 0.25.x and 0.26.x
+# - 0.27.2 can build 0.27.x but not 0.25.x, 0.26.x and 0.29.x
```

Comment:
I'm intrigued by how crystal is starting to become maintained in nixpkgs.

Is there are stable change to be done once a new version is released?

Usually the last `0.m.p` version should be used to build any `0.(m+1).q` version. That is how Crystal releases works. That is the rule that should be followed since the compiler is bootstrapped. Although in the past there were some hiccups with this rule.

In brew, when `0.m.p` is released we built that tag using the GitHub binary release of the latest `0.m-1.q` version.
But from what I "understand" from the formula after a first version of the compiler got into nix the new ones are built after it. Having a new lineage. Am I right?